syncano/connection.py

```diff
@@ -181,10 +182,26 @@ def make_request(self, method_name, path, **kwargs):
         # Encode request payload
         if 'data' in params and not isinstance(params['data'], six.string_types):
             params['data'] = json.dumps(params['data'])
-
         url = self.build_url(path)
         response = method(url, **params)
+        content = self.get_response_content(url, response)
+
+        if files:
+            params.pop('data')
+            params['headers'].pop('content-type')
```

Comment:
Because `content-type` indicates we send data as serialized json and indeed `data` is serialized json. As I wrote in description, file object isn't serializable. Furthermore, leaving `data` in params gives **ValueError: Data must not be a string.** That's why I propose second request.